src/app/icd/server/ICDCheckInSender.cpp

```diff
@@ -27,10 +27,14 @@ namespace app {
 
 using namespace Protocols::SecureChannel;
 
-ICDCheckInSender::ICDCheckInSender(Messaging::ExchangeManager * exchangeManager)
+ICDCheckInSender::ICDCheckInSender(Messaging::ExchangeManager * exchangeManager, AddressResolve::Resolver * addressResolver)
```

Comment:
These changes for the address resolver seem unrelated to ClientType.  Shouldn't they be in a separate PR?

Reply:
done